dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/logger/LogcatLogHandler.kt

```diff
@@ -70,14 +70,22 @@ internal class LogcatLogHandler(
     internal fun getCallerStackElement(): StackTraceElement? {
         return if (Datadog.isDebug && useClassnameAsTag) {
             val stackTrace = Throwable().stackTrace
-            stackTrace.firstOrNull {
-                it.className !in ignoredClassNames
-            }
+            return findValidCallStackElement(stackTrace)
         } else {
             null
         }
     }
 
+    internal fun findValidCallStackElement(
+        stackTrace: Array<StackTraceElement>
+    ): StackTraceElement? {
+        return stackTrace.firstOrNull {
+            it.className !in ignoredClassNames &&
+                !it.className.startsWith("com.datadog.android.timber") &&
```

Comment:
`startsWith` instead of `contains` to be more restrictive, since the latter can lead to false positives.